src/verifai/scenic_server.py

```diff
@@ -23,25 +30,29 @@ def __init__(self, sampling_data, monitor, options={}):
         else:
             self.rejectionFeedback = extSampler.rejectionFeedback
         self.monitor = monitor
+        if isinstance(self.monitor, multi_objective_monitor):
+            self.sampler.scenario.externalSampler.sampler.domainSampler.split_sampler.samplers[0].set_graph(self.monitor.graph)
         self.lastValue = None
         defaults = DotMap(maxSteps=None, verbosity=0, maxIterations=1)
         defaults.update(options)
         self.maxSteps = defaults.maxSteps
         self.verbosity = defaults.verbosity
         self.maxIterations = defaults.maxIterations
-        # TODO call super after refactoring networking into Server subclass?
 
     def run_server(self):
+        start = time.time()
         sample = self.sampler.nextSample(self.lastValue)
         scene = self.sampler.lastScene
         assert scene
+        after_sampling = time.time()
         result = self._simulate(scene)
         if result is None:
             self.lastValue = self.rejectionFeedback
         else:
             self.lastValue = (0 if self.monitor is None
-                              else self.monitor.evaluate(result.trajectory))
-        return sample, self.lastValue
+                              else self.monitor.evaluate(result))
```

Comment:
Why pass the entire `Simulation` to the monitor? Monitors expect a trajectory - has that changed?